lib/rules/no-wait-for-snapshot.ts

```diff
@@ -0,0 +1,115 @@
+
+export const RULE_NAME = 'no-wait-for-snapshot';
+export type MessageIds = 'noWaitForSnapshot';
+type Options = [];
+
+const ASYNC_UTILS_REGEXP = new RegExp(`^(${ASYNC_UTILS.join('|')})$`);
+
+export default ESLintUtils.RuleCreator(getDocsUrl)<Options, MessageIds>({
+  name: RULE_NAME,
+  meta: {
+    type: 'problem',
+    docs: {
+      description:
+        'Ensures no snapshot is generated inside of a `waitFor` call',
+      category: 'Best Practices',
+      recommended: 'warn',
+    },
+    messages: {
+      noWaitForSnapshot:
+        "A snapshot can't be generated inside of a `waitFor` call",
+    },
+    fixable: null,
+    schema: [],
+  },
+  defaultOptions: [],
+
+  create(context) {
+    const asyncUtilsUsage: Array<{
+      node: TSESTree.Identifier | TSESTree.MemberExpression;
+      name: string;
+    }> = [];
+    const importedAsyncUtils: string[] = [];
+    const snapshotUsage: TSESTree.Identifier[] = [];
+
+    return {
+      'ImportDeclaration > ImportSpecifier,ImportNamespaceSpecifier'(
+        node: TSESTree.Node
+      ) {
+        const parent = node.parent as TSESTree.ImportDeclaration;
+
+        if (!LIBRARY_MODULES.includes(parent.source.value.toString())) return;
+
+        let name;
+        if (node.type === 'ImportSpecifier') {
+          name = node.imported.name;
+        }
+
+        if (node.type === 'ImportNamespaceSpecifier') {
+          name = node.local.name;
+        }
+
+        importedAsyncUtils.push(name);
+      },
+      [`CallExpression > Identifier[name=${ASYNC_UTILS_REGEXP}]`](
+        node: TSESTree.Identifier
+      ) {
+        asyncUtilsUsage.push({ node, name: node.name });
+      },
+      [`CallExpression > MemberExpression > Identifier[name=${ASYNC_UTILS_REGEXP}]`](
+        node: TSESTree.Identifier
+      ) {
+        const memberExpression = node.parent as TSESTree.MemberExpression;
+        const identifier = memberExpression.object as TSESTree.Identifier;
+        const memberExpressionName = identifier.name;
+
+        asyncUtilsUsage.push({
+          node: memberExpression,
+          name: memberExpressionName,
+        });
+      },
+      [`Identifier[name='toMatchSnapshot']`](node: TSESTree.Identifier) {
+        snapshotUsage.push(node);
+      },
+      'Program:exit'() {
+        const testingLibraryUtilUsage = asyncUtilsUsage.filter(usage => {
+          if (usage.node.type === 'MemberExpression') {
+            const object = usage.node.object as TSESTree.Identifier;
+
+            return importedAsyncUtils.includes(object.name);
+          }
+
+          return importedAsyncUtils.includes(usage.name);
+        });
+
+        function isWithinAsyncUtil(
+          asyncUtilUsage: {
+            node: TSESTree.Identifier | TSESTree.MemberExpression;
+            name: string;
+          },
+          node: TSESTree.Node
+        ) {
+          let callExpression = findClosestCallExpressionNode(node);
+          while (callExpression != null) {
```

Comment:
I'd have hoped the linter configuration enforced `!==` over `!=`

I tend to avoid `!=` completely but I'm not sure here 😅 

Thoughts @Belco90 ?

Reply:
I think the linter allows for `!=` because I'm checking against `null`. If you use `!=` it's implicitly checking for `undefined` too

Reply:
I think in this case is fine to compare against nil, and I understand that's the actual usage right?